examples/custom_backend/mlkem_native/fips202/native/custom/src/sha3.c

```diff
@@ -31,19 +31,25 @@ void sha3_keccakf(uint64_t st[25])
   int i, j, r;
   uint64_t t, bc[5];
 
-#if __BYTE_ORDER__ != __ORDER_LITTLE_ENDIAN__
-  uint8_t *v;
-
-  /* endianess conversion. this is redundant on little-endian targets */
-  for (i = 0; i < 25; i++)
-  {
-    v = (uint8_t *)&st[i];
-    st[i] = ((uint64_t)v[0]) | (((uint64_t)v[1]) << 8) |
-            (((uint64_t)v[2]) << 16) | (((uint64_t)v[3]) << 24) |
-            (((uint64_t)v[4]) << 32) | (((uint64_t)v[5]) << 40) |
-            (((uint64_t)v[6]) << 48) | (((uint64_t)v[7]) << 56);
-  }
-#endif
+  /* NOTE:
+   * This is present in the tiny_sha3 implementation because it uses
+   * a byte-reversed presentation of the Keccakf1600 state for big endian
+   * targets. mlkem-native uses the standard presentation, hence we don't
+   * the reversal here. */
+  /* #if __BYTE_ORDER__ != __ORDER_LITTLE_ENDIAN__ */
+  /*   uint8_t *v; */
+
+  /*   /\* endianess conversion. this is redundant on little-endian targets *\/
+   */
+  /*   for (i = 0; i < 25; i++) */
+  /*   { */
+  /*     v = (uint8_t *)&st[i]; */
+  /*     st[i] = ((uint64_t)v[0]) | (((uint64_t)v[1]) << 8) | */
+  /*             (((uint64_t)v[2]) << 16) | (((uint64_t)v[3]) << 24) | */
+  /*             (((uint64_t)v[4]) << 32) | (((uint64_t)v[5]) << 40) | */
+  /*             (((uint64_t)v[6]) << 48) | (((uint64_t)v[7]) << 56); */
+  /*   } */
+  /* #endif */
```

Comment:
I'm not quite sure I understand. 
Why does tiny_sha3 do it in the first place, if you are able to just remove it?
Is it because adding into the state and extracting from the state is more efficient from the byte-reversed representation and that's outweighs the cost of the reversal in the permutation itself?